Backend/AlgorithmsManager.py

```diff
@@ -468,6 +473,64 @@ def get_most_frequent_retweeters(self, username, num_tweets=20, num_retweets=20)
 
         return res
 
+    def get_network_tweets_text(self, id2, username, count, depth):
+        textsStr = ""
+        if id2 > 0 or username is not None:
+            # get bunch of tweets starting with the user of given id
+            n = depth  # goes depth level deep
+            # array of tweet texts
+            texts = []
+            self.append_texts_recursive(texts, count, n, 0, id2, username)
+
+            for text in texts:
+                splittext = text.split(' ')
+                for split in splittext:
+                    if not split.startswith('http'):
+                        textsStr += (split + " ")
+        return textsStr
+
+    def append_texts_recursive(self, texts, count, n, i, id2, username):
+        tweets = self.twitter.getTweetsFromUser(id2, username, count)
+        for text in tweets:
+            texts.append(text)
+        if i < n:
+            followsids = self.twitter.getFollowersID(id2, username, count)
+            for id3 in followsids:
+                screen_name = self.twitter.get_username_from_id(id3)
+                self.append_texts_recursive(texts, n, count, i+1, id3, screen_name)
```

Comment:
just as a note for later (not an issue for this pull request), if get_network_tweets_text is having speed problems it might be worth trying to re-implement this logic using a while loop and a stack data structure rather than using a recursive algorithm.

Reply:
Also, I'd expect this to be really slow unless depth was 1-2 (maaaaaybe 3?). This is some really impressively aggressive scraping- I thought we'd only been talking about grabbing the tweets of a user's followers (and this does that given a depth of 1), but you did that and more!

Reply:
Yes definitely better data structure, threading, and loop usage will dramatically boost the performance